core/src/main/java/com/datastax/oss/driver/internal/core/channel/DefaultWriteCoalescer.java

```diff
@@ -84,8 +94,10 @@ private Flusher(EventLoop eventLoop) {
 
     private void enqueue(Write write) {
       boolean added = writes.offer(write);
+      System.out.println("writes size: " + writes.size() + " for Flusher: " + this);
       assert added; // always true (see MpscLinkedAtomicQueue implementation)
-      if (running.compareAndSet(false, true)) {
+      if (write.isWritable() && running.compareAndSet(false, true)) {
+        System.out.println("runOnEventLoop#100");
         eventLoop.execute(this::runOnEventLoop);
```

Comment:
The problem is that there is no rescheduling here: if `!write.isWritable()`, the writes will simply be left pending.

Of course if another request is executed after the channel becomes writable again, it will dequeue all the accumulated writes. But this is not guaranteed to happen; if it doesn't the requests will hang forever.

Reply:
yes, that is why we need to register on the `channelWritabilityChanged()` callback - it signals that `isWritable()` changed; in such a case we should start event loop

Reply:
I've propagated the `channelWritabilityChanged` in the: https://github.com/datastax/java-driver/pull/1465/commits/341e7d7927e6f0021094c1324386f94d998e4261 - please take a look. The requests are no left pending - when the buffer is able to accept the messages, the event loop is restarted